keyboards/handwired/frame_48/rev1/rules.mk

```
@@ -0,0 +1,5 @@
+# MCU name
+MCU = atmega32u4
+BOOTLOADER = caterina
+ENCODER_MAP_ENABLE = yes
+CONVERT_TO=kb2040
```

Encoder map should not be enabled by default. 

Also, shouldn't use the converter here. Either it should be Pro Micro supported, or it should be RP2040, and use the RP2040 specific pin definitions. 

```suggestion
MCU = RP2040
BOARD = QMK_PM2040
BOOTLOADER = rp2040
```

---

Hey! I got all the pin definitions added but looks like I am getting an error relating to the serial connection? I'm having trouble finding an example of how this works for the kb2040 board, and if I need to specify a different serial driver or something? Any ideas here? Thanks so much for your help!

`QMK Firmware 0.18.15
Making handwired/frame_48/rev1 with keymap default

arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 8-2019-q3-update) 8.3.1 20190703 (release) [gcc-8-branch revision 273027]
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

Compiling: quantum/command.c                                                                        [OK]
Compiling: platforms/chibios/drivers/serial.c                                                      platforms/chibios/drivers/serial.c: In function 'Thread1':
platforms/chibios/drivers/serial.c:90:9: error: implicit declaration of function 'palWaitLineTimeout'; did you mean 'oqWriteTimeout'? [-Werror=implicit-function-declaration]
         palWaitLineTimeout(SOFT_SERIAL_PIN, TIME_INFINITE);
         ^~~~~~~~~~~~~~~~~~
         oqWriteTimeout
platforms/chibios/drivers/serial.c: In function 'soft_serial_target_init':
platforms/chibios/drivers/serial.c:103:5: error: implicit declaration of function 'palEnablePadEvent' [-Werror=implicit-function-declaration]
     palEnablePadEvent(PAL_PORT(SOFT_SERIAL_PIN), PAL_PAD(SOFT_SERIAL_PIN), PAL_EVENT_MODE_FALLING_EDGE);
     ^~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
 [ERRORS]
 | 
 | 
 | 
make[1]: *** [.build/obj_handwired_frame_48_rev1_default/serial.o] Error 1
make: *** [handwired/frame_48:default] Error 1
Make finished with errors`

---

You might have to set the SERIAL_DRIVER = vendor as stated here:

https://docs.qmk.fm/#/serial_driver?id=setup-1

if it ask you for the vendor PID VID here's the relevent json section:

    "usb": {
        "device_version": "1.0.0",
        "pid": "0x0000",
        "vid": "0xFEED"
    }

---

Wooo! Thanks so much! That worked.